### PR TITLE
ProblemSpec.withException() accepts Throwable

### PR DESCRIPTION
**Pull request by donat**
Fixes https://github.com/gradle/gradle/issues/30565

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

**Review by hegyibalint — APPROVED** [review]
LGTM, but I would've guessed this would make the compiler cry about the checked exception. Apparently not.

Have any idea what's going on here?

**Comment by donat** [comment]
@hegyibalint because the PR only changes the exception type associated with the report. The `throwing()` method still throws a RuntimeEsxception:

https://github.com/gradle/gradle/pull/30628/files#diff-35517c5c4d1db63123d802837a01ae794d5cfbc0782a31d48adf18bea5df5a1cR73-R77

**Comment by donat** [comment]
@bot-gradle merge

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemReporter.java [hunk 1]
```diff
@@ -55,22 +55,26 @@ public void reporting(Action<ProblemSpec> spec) {
     }
 
     @Override
-    public RuntimeException throwing(Action<ProblemSpec> spec) {
+    public RuntimeException throwing(Action<ProblemSpec> spec)  {
```
Comment (MGaetan89) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemReporter.java:58]:
```suggestion
    public RuntimeException throwing(Action<ProblemSpec> spec) {
```

**Pull request merged**